packages/flutter_tools/test/ios/cocoapods_test.dart

```diff
@@ -132,55 +134,65 @@ void main() {
     }, overrides: <Type, Generator>{
       FileSystem: () => fs,
     });
 
-    testUsingContext('creates swift Podfile if swift', () {
+    testUsingContext('creates swift Podfile if swift', () async {
       when(mockXcodeProjectInterpreter.isInstalled).thenReturn(true);
       when(mockXcodeProjectInterpreter.getBuildSettings(any, any)).thenReturn(<String, String>{
         'SWIFT_VERSION': '4.0',
       });
 
-      cocoaPodsUnderTest.setupPodfile('project');
+      final FlutterManifest manifest =
+          await FlutterManifest.createFromPath(fs.path.join('project', bundle.defaultManifestPath));
+      cocoaPodsUnderTest.setupPodfile('project', manifest);
 
       expect(podFile.readAsStringSync(), 'Swift podfile template');
     }, overrides: <Type, Generator>{
       FileSystem: () => fs,
       XcodeProjectInterpreter: () => mockXcodeProjectInterpreter,
     });
 
-    testUsingContext('does not recreate Podfile when already present', () {
+    testUsingContext('does not recreate Podfile when already present', () async {
       podFile..createSync()..writeAsStringSync('Existing Podfile');
 
-      cocoaPodsUnderTest.setupPodfile('project');
+      final FlutterManifest manifest =
+          await FlutterManifest.createFromPath(fs.path.join('project', bundle.defaultManifestPath));
+      cocoaPodsUnderTest.setupPodfile('project', manifest);
 
       expect(podFile.readAsStringSync(), 'Existing Podfile');
     }, overrides: <Type, Generator>{
       FileSystem: () => fs,
     });
 
-    testUsingContext('does not create Podfile when we cannot interpret Xcode projects', () {
+    testUsingContext('does not create Podfile when we cannot interpret Xcode projects', () async {
       when(mockXcodeProjectInterpreter.isInstalled).thenReturn(false);
 
-      cocoaPodsUnderTest.setupPodfile('project');
+      final FlutterManifest manifest =
+          await FlutterManifest.createFromPath(fs.path.join('project', bundle.defaultManifestPath));
+      cocoaPodsUnderTest.setupPodfile('project', manifest);
 
       expect(podFile.existsSync(), false);
     }, overrides: <Type, Generator>{
       FileSystem: () => fs,
       XcodeProjectInterpreter: () => mockXcodeProjectInterpreter,
     });
 
-    testUsingContext('includes Pod config in xcconfig files, if not present', () {
+    testUsingContext('includes Pod config in xcconfig files, if not present', () async {
       podFile..createSync()..writeAsStringSync('Existing Podfile');
       debugConfigFile..createSync(recursive: true)..writeAsStringSync('Existing debug config');
       releaseConfigFile..createSync(recursive: true)..writeAsStringSync('Existing release config');
 
-      cocoaPodsUnderTest.setupPodfile('project');
+      final FlutterManifest manifest =
+          await FlutterManifest.createFromPath(fs.path.join('project', bundle.defaultManifestPath));
+      cocoaPodsUnderTest.setupPodfile('project', manifest);
```

Comment:
Reduce duplication of how we create a manifest?

BTW, I suppose we could make the `FlutterManifest.createFromPath` synchronous, if it simplifies things.

Reply:
The schema can only be created async. Therefore FlutterManifest.createFromPath cannot be sync.
I have put the logic of loading the manifest into FlutterProject.